### PR TITLE
psy10 update

### PR DESCRIPTION
**Pull request by jd-lara**
@daniel-thom I tagged you to see the adaptation to the newer scheme in PSY for the dynamic data

### CODE REVIEW

File: src/base/simulation.jl [hunk 1]
```diff
@@ -249,27 +249,28 @@ function _attach_ports!(component::PSY.DynamicComponent)
 end
 
 function _attach_inner_vars!(
-    device::PSY.DynamicGenerator,
+    dyn_data::PSY.DynamicGenerator,
```
Comment (daniel-thom) [inline src/base/simulation.jl:252]: I'm sure you changed this name for a reason, but it reads worse for me. `data` seems too generic.

File: src/base/simulation_inputs.jl [hunk 2]
```diff
@@ -125,11 +128,21 @@ function build!(inputs::SimulationInputs)
...
+
+    if isempty(dynamic_injection)
         error("System doesn't contain any DynamicInjection devices")
+    end
+
     for d in dynamic_injection
-        if !(:states in fieldnames(typeof(d)))
+        @debug PSY.get_name(d)
+        dyn_data = PSY.get_dynamic_injector(d)
+        if !(:states in fieldnames(typeof(dyn_data)))
```
Comment (daniel-thom) [inline src/base/simulation_inputs.jl:145]: Could this behavior be expressed in a function rather than requiring a specific field name?

Comment (jd-lara) [reply src/base/simulation_inputs.jl]: We will require that all DynamicInjection in PSY implements `get_states`

**Review by rodrigomha — APPROVED** [review]
LGTM

**Pull request merged**